homeassistant/components/roborock/config_flow.py

```diff
@@ -91,6 +99,21 @@ async def async_step_code(
                 _LOGGER.exception(ex)
                 errors["base"] = "unknown"
             else:
+                if self.source == SOURCE_REAUTH:
+                    entry = self.hass.config_entries.async_get_entry(
```

Comment:
A common approach I have seen emerging in recent PRs is to just track the entry when re-auth starts then update that same entry in reauth confirm:
```
https://github.com/home-assistant/core/blob/9c5e0fc2c93c597510ddacee9c31ea988c960ac2/homeassistant/components/google_mail/config_flow.py#L26
```
I guess it just avoids a lookup here.

Reply:
fair enough - I had seen that approach, but it did seem a bit odd to me as it is just adding a member variable that I feel like isn't needed and it still makes the same call? But I have changed it anyways because it does seem the most common.

Reply:
Agree its a minor point.  I think maybe the only benefit i see is that it pushes the lookup to the start, then you can use the existence of the entry as the "are we doing reauth" check.  Not a major point.